astropy/convolution/__init__.py

```diff
@@ -1,11 +1,19 @@
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
 
 
-from .core import *  # noqa
-from .kernels import *  # noqa
-from .utils import discretize_model, KernelSizeError  # noqa
+from . import core, kernels
+from .convolve import (convolve, convolve_fft, convolve_models,
+                       convolve_models_fft, interpolate_replace_nans)
+from .core import *
+from .kernels import *
+from .kernels import MexicanHat1DKernel, MexicanHat2DKernel  # Deprecated kernels
+from .utils import *
 
-from .convolve import convolve, convolve_fft, interpolate_replace_nans, convolve_models, convolve_models_fft  # noqa
-
-# Deprecated kernels that are not defined in __all__
-from .kernels import MexicanHat1DKernel, MexicanHat2DKernel
+__all__ = (
+    ["convolve", "convolve_fft", "convolve_models", "convolve_models_fft",
+     "interpolate_replace_nans"]
+    + core.__all__
+    + kernels.__all__
+    + ["MexicanHat1DKernel", "MexicanHat2DKernel"]
+    + utils.__all__
+)
```

Comment:
I don't think we should do this - in general we have followed the pattern of not having __all__ defined in top level unit files, and if we are going to programmatically populate it like this it really serves no purpose?

Reply:
I changed my review away from 'approved' b/c of this - I don't understand the broader implications of the proposed change.

Reply:
I will remove it.
I personally like having ``__all__`` defined at every level to never break expectations — ``__all__`` is always defined.

Reply:
Removed.